src/adapters/surrealDB/index.ts

```diff
@@ -0,0 +1,189 @@
+import { trainCase } from 'case-anything';
+import type { Pipeline } from '../../types/pipeline/base';
+import type { EnrichedBormSchema, PipelineOperation } from '../../types';
+import { enrichBQLQuery } from '../../pipeline/preprocess/query/enrichBQLQuery';
+import { cleanQueryRes } from './pipeline/postprocess/query/cleanQueryRes';
+import { pascal, dash } from 'radash';
+import { QueryPath } from '../../types/symbols';
+import { produce } from 'immer';
+import type {
+	SurrealDbResponse,
+	EnrichedBqlQuery,
+	EnrichedBqlQueryEntity,
+	EnrichedBqlQueryAttribute,
+} from './types/base';
+
+const getSubtype = (
+	schema: EnrichedBormSchema,
+	kind: 'entities' | 'relations',
+	thing: string,
+	result: Array<string> = [],
+): Array<string> => {
+	const subtypes = Object.values(schema[kind])
+		.filter((itemSchema) => itemSchema.extends === thing)
+		.map((itemSchema) => itemSchema.name);
+
+	if (subtypes.length === 0) {
+		return result;
+	}
+
+	for (const subtype of subtypes) {
+		result.push(subtype);
```

Comment:
Is there a way to do this in an immutable way? We tend to avoid `let` variables as much as possible and use immer when we don't have the choice.

Also it is completely fine to use let when there is not an easy immutable alternative, we were forced to do it a lot of times unfortunately

Reply:
No problem, I have now rewritten it with `Array.reduce`